### PR TITLE
Reworking to beans

### PR DESCRIPTION
**Pull request by JJerome-NM**
1. Transferred the project from multi-module to standard
2. Created conifiguration for the correct start of the web socket application
3. Simplified the inclusion of web sockets
4. Initialization is reworked under beans for the correct injection of components into the fields of the class controller

### CODE REVIEW

File: src/main/java/com/jjerome/annotations/SocketController.java [hunk 1]
```diff
@@ -1,10 +1,13 @@
 package com.jjerome.annotations;
 
+import org.springframework.stereotype.Component;
+
 import java.lang.annotation.*;
```
Comment (justme1135) [inline src/main/java/com/jjerome/annotations/SocketController.java:5]: Rewrite to single imports

File: src/main/java/com/jjerome/models/BeanUtil.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    private final static Logger LOGGER = LoggerFactory.getLogger(BeanUtil.class);
+
+    private final ApplicationContext context;
+
+
+    public Set<Class<?>> findSpringBootApplicationBeanClass(){
+        String[] beanNames = this.context.getBeanNamesForAnnotation(SpringBootApplication.class);
+
+        if (beanNames.length > 1) {
+            LOGGER.warn("OMGGGGGGGGGGGGGGG You have more than two SpringBootApplications, this can cause errors");
```
Comment (justme1135) [inline src/main/java/com/jjerome/models/BeanUtil.java:26]: Really funny but please rewrite to more meaningful message.

File: src/main/java/com/jjerome/context/SocketControllersContext.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        if (!(method.getGenericParameterTypes()[0] instanceof ParameterizedType parameterizedType)) return null;
+
+        return (Class<?>) parameterizedType.getActualTypeArguments()[0];
+    }
+
+    public boolean validateMappingMethod(Method method, Class<? extends Annotation> methodAnnotation,
+                                         Class<?> returnClass, Class<?>... parameterClasses){
+        if (!method.isAnnotationPresent(methodAnnotation)) {
+            return false;
+        } else if (method.getParameterCount() != parameterClasses.length) {
```
Comment (justme1135) [inline src/main/java/com/jjerome/context/SocketControllersContext.java:59]: not need for `else` if you are already using `return` in previous condition

**Review by justme1135 — CHANGES_REQUESTED** [review]

File: src/main/java/com/jjerome/annotations/SocketController.java (hunk 1 above)
Comment (JJerome-NM) [reply src/main/java/com/jjerome/annotations/SocketController.java]: Fixed

File: src/main/java/com/jjerome/models/BeanUtil.java (hunk 2 above)
Comment (JJerome-NM) [reply src/main/java/com/jjerome/models/BeanUtil.java]: I redesigned it to make it more informative

File: src/main/java/com/jjerome/context/SocketControllersContext.java (hunk 3 above)
Comment (JJerome-NM) [reply src/main/java/com/jjerome/context/SocketControllersContext.java]: Yes you are right, I fixed it

**Review by justme1135 — APPROVED** [review]

**Pull request merged**